src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -207,7 +207,7 @@ pub struct HostDevice {
     pub vfio_type: VfioDeviceType,
 
     /// guest PCI path of device
-    pub guest_pci_path: Option<PciPath>,
+    pub guest_pci_path: PciPath,
```

Comment:
I was giving this more thought..do we really need to remove the Option here. The `guest_pci_path` may be unitialised in some cases, for example qemu currently does not set `guest_pci_path` as that code is not implemented yet. In general I think since `guest_pci_path` may be unitialised/unassigned by the hypervisor I think it may be a good idea to keep this as an option.
Would like to hear what @jodh-intel and others think about this as well.

Reply:
@amshinde I don't think so. 
It's an incontestable fact that `guest_pci_path` exists in Qemu, CLH, and Dragonball, regardless of VMM support， right? 
For CLH/dragonball vfio device, what does it mean if `guest_pci_path` uses Option? Is there a case where pci path does not exist? IMO, Option is not set for whether some VMM exists. It is set for whether `guest_pci_path` exists in the VMM.
I want this field `guest_pci_path` to have a definite meaning. The word "Option" gives a misleading impression, so I want to try to remove it.

The `guest_pci_path` is initialized with default value, and Rust does not allow uninitialized or unassigned variables to be used. This is guaranteed by its memory safety.

Reply:
CC @jodh-intel

Reply:
I agree with @amshinde  - if there is any chance `guest_pci_path` is not set by a hypervisor (unrelated to rust variable initialisation), then we should keep the `Option` imho as that could be a nasty source of bugs. Maybe @pmores can comment from the QEMU driver perspective?

Reply:
@jodh-intel Yes I agree with the point that the `guest_pci_path` is unrelated to rust variable initialisation.

Reply:
@jodh-intel @amshinde But I’m curious if Qemu can do without `guest_pci_path`. Or is there a new way to work around ? 
AFAIK, regardless of the VMM, the same kata-agent code will be used, right?
For VFIO devices, we also need to do the mapping between the host device BDF and the PCI path that will be generated after it is inserted into the VM on the runtime side. Then, when building the kata agent device, we assign the mapping relationship to `Device.options`, and then pass it to the kata agent and parse the mapping relationship.

If Qemu can avoid do this,   or `guest_pci_path` is no point for it. Ok, I will agree with Option<PciPath>

Reply:
> @jodh-intel @amshinde But I’m curious if Qemu can do without `guest_pci_path`. Or is there a new way to work around ? And if we're all concern about the Option and have difference points about it, I will remove the commit about removing Option. We can disscus it when Qemu does implement the vfio feature.

We're not there yet but I agree with @Apokleos. The QEMU driver in the go runtime does set it :

```
                device.GuestPciPath, err = q.arch.qomGetPciPath(device.ID, &q.qmpMonitorCh)
```

I see no reason for the runtime-rs driver to behave differently.